src/PlayerSkeleton.java

```diff
@@ -25,3 +26,204 @@ public static void main(String[] args) {
 	}
 	
 }
+
+class Heuristics {
+	private int lineCleared;
+	
+	private double weightOfLineCleared;
+	
+	private int numberOfHoles;
+	
+	private double weightOfNumberOfHoles;
+	
+	private int totalWeightOfHoles;
+	
+	private double weightOfTotalWeightOfHoles;
+	
+	private int sumOfAdjacentColumnHeightDifference;
+	
+	private double weightOfSumOfAdjacentColumnHeightDifference;
+	
+	private int landingHeight;
+	
+	private double weightOfLandingHeight;
+	
+	private int totalHeight;
+	
+	private double weightOfTotalHeight;
+	
+	private int rangeOfHeight;
+	
+	private double weightOfRangeOfHeight;
+	
+	
+	public Heuristics(State thisState, State nextState) {
+		this.lineCleared = nextState.getRowsCleared();
+		

```

Comment:
perhaps you could abstract out these operations into individual functions like `getNumberOfHoles(State state)`